modules/renter/contractor/editor.go

```diff
@@ -84,216 +59,68 @@ func (he *hostEditor) EndHeight() types.BlockHeight { return he.contract.FileCon
-		return errors.New("contract has already ended")
-	}
-
-	// calculate price
-	sectorPrice, sectorCollateral := types.ZeroCurrency, types.ZeroCurrency
-
-	// calculate the new Merkle root
-	newRoots := make([]crypto.Hash, 0, len(he.contract.MerkleRoots))
-	index := -1
-	for i, h := range he.contract.MerkleRoots {
-		if h == root {
-			index = i
-		} else {
-			newRoots = append(newRoots, h)
-		}
-	}
-	if index == -1 {
-		return errors.New("no record of that sector root")
+	contract, err := he.editor.Delete(root)
+	if err != nil {
+		return err
 	}
-	merkleRoot := cachedMerkleRoot(newRoots)
 
-	// create the action and accompanying revision
-	actions := []modules.RevisionAction{{
-		Type:        modules.ActionDelete,
-		SectorIndex: uint64(index),
-	}}
-	rev := newRevision(he.contract.LastRevision, merkleRoot, uint64(len(newRoots)), sectorPrice, sectorCollateral)
+	he.contractor.mu.Lock()
+	he.contractor.contracts[contract.ID] = contract
+	he.contractor.saveSync()
+	he.contractor.mu.Unlock()
+	he.contract = contract
 
-	// run the revision iteration
-	return he.runRevisionIteration(actions, rev, newRoots, height)
+	return nil
 }
 
 // Modify negotiates a revision that edits a sector in a file contract.
 func (he *hostEditor) Modify(oldRoot, newRoot crypto.Hash, offset uint64, newData []byte) error {
-	if he.stopped {
-		return modules.ErrStopResponse
-	}
-	// allot 10 minutes for this exchange; sufficient to transfer 4 MB over 50 kbps
-	he.conn.SetDeadline(time.Now().Add(600 * time.Second))
-	defer he.conn.SetDeadline(time.Now().Add(time.Hour)) // reset deadline
-
-	// get current height
-	he.contractor.mu.RLock()
-	height := he.contractor.blockHeight
-	he.contractor.mu.RUnlock()
-	if height >= he.contract.FileContract.WindowStart {
-		return errors.New("contract has already ended")
-	}
-
-	// calculate price
-	sectorPrice := he.host.UploadBandwidthPrice.Mul64(uint64(len(newData)))
-	if sectorPrice.Cmp(he.contract.LastRevision.NewValidProofOutputs[0].Value) >= 0 {
-		return errors.New("contract has insufficient funds to support upload")
-	}
-
-	// calculate the new Merkle root
-	newRoots := make([]crypto.Hash, len(he.contract.MerkleRoots))
-	index := -1
-	for i, h := range he.contract.MerkleRoots {
-		if h == oldRoot {
-			index = i
-			newRoots[i] = newRoot
-		} else {
-			newRoots[i] = h
-		}
-	}
-	if index == -1 {
-		return errors.New("no record of that sector root")
-	}
-	merkleRoot := cachedMerkleRoot(newRoots)
-
-	// create the action and revision
-	actions := []modules.RevisionAction{{
-		Type:        modules.ActionModify,
-		SectorIndex: uint64(index),
-		Offset:      offset,
-		Data:        newData,
-	}}
-	rev := newModifyRevision(he.contract.LastRevision, merkleRoot, sectorPrice)
-
-	// run the revision iteration
-	if err := he.runRevisionIteration(actions, rev, newRoots, height); err != nil {
+	oldUploadSpending := he.editor.UploadSpending
+	contract, err := he.editor.Modify(oldRoot, newRoot, offset, newData)
```

Comment:
Might want to name `Modify` to `Revise` to keep parity with the rest of the codebase.